pylint/extensions/magic_value.py

```diff
@@ -74,10 +98,23 @@ def _check_constants_comparison(self, node: nodes.Compare) -> None:
                 confidence=HIGH,
             )
 
-    def _is_magic_value(self, node: nodes.NodeNG) -> bool:
+    def _is_magic_value(self, node: nodes.Const) -> bool:
         return (not utils.is_singleton_const(node)) and (
-            node.value not in self.linter.config.valid_magic_values
+            node.value not in (self.valid_magic_vals)
+        )
+
+    @staticmethod
+    def _parse_rcfile_magic_numbers(parsed_val: str) -> float | str:
```

Comment:
This feels like it should in the configuration parsing instead of individual checkers. Have you tried if that breaks stuff?

Reply:
Good idea it'd be much better, I thought this is a behavior that is not useful for other checkers initially.. I would have a try though.
Are all checkers with `pylintrc` config vars have functional tests for regression? I wouldn't be surprised if not...

Reply:
I think it makes sense to do this for all `pylintrc` files. We'll get a report about it if we break anything :smile:

Reply:
Added this to `argument_manager.py` in `_parse_configuration_file` method in the begining (before the parser to `self.config`):

```
            for idx,arg in enumerate(arguments):
                if "--" not in arg:
                    arguments[idx] = ','.join([self._post_parse_arg(val) for val in arg.strip(',').replace(" ", "").split(',')])
```

running on `magic_value` test you get, as input to this method:

```
>>>arguments
['--load-plugins', 'pylint.extensions.magic_value,', '--valid-magic-values', "0, 1, -1, __main__,'', \\n, \\\\b"]
```

and after this snippet:

```
>>>arguments
['--load-plugins', 'pylint.extensions.magic_value', '--valid-magic-values', '0,1,-1,__main__,,\n,\\b']
```

which is kind of the wanted behavior (Except for maybe the empty string). but after:

```
            self.config, parsed_args = self._arg_parser.parse_known_args(
                arguments, self.config
            )
```

you get:

```
>>>self.config.valid_magic_values
['0', '1', '-1', '__main__', '\\b']
```

which kills the empty string and newline character. 

So I am not sure how to tackle this, I can't control the argparse package..
If it should be done, I would do it in another commit, right now uploading in `magic_value.py`

Reply:
this is the extra method for comletion:

  ```
  @staticmethod
    def _post_parse_arg(parsed_val: str) -> float | str:
        parsed_val = parsed_val.encode().decode('unicode_escape')

        if parsed_val.startswith("'") and parsed_val.endswith("'"):
            return parsed_val[1:-1]

        return parsed_val
```